haigha/transports/socket_transport.py

```diff
@@ -29,23 +29,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
+        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
+        if not infos:
+            raise socket.error("getaddrinfo returns an empty list")
+
+        # Sort addresses such that the new IPv6 address is first.
+        for info in reversed(sorted(infos, key=lambda info: len(info[-1]))):
+
+            sockaddr = info[-1]
+            try:
+                # Check if the IP is a valid IPv6 address. Fallback to IPv4 if not.
+                socket.inet_pton(socket.AF_INET6, sockaddr[0])
```

Comment:
The purpose of this section is to determine the appropriate family with which to initialize the socket object. If the current address in question is a valid IPv6 address then we have to initialize the socket with `AF_INET6` or it will fail to connect.

An alternative to the try/catch and use of `inet_pton` is to, instead, check the length of the `sockaddr` variable which would be length 2 for IPv4 and 4 for IPv6.